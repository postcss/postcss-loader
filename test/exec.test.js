import path from 'path';

import {
  compile,
  getCompiler,
  getErrors,
  getCodeFromBundle,
  getWarnings,
} from './helpers';

describe('"exec" option', () => {
  it('should work with "Boolean" value', async () => {
    const compiler = getCompiler(
      './jss/exec/index.js',
      {},
      {
        module: {
          rules: [
            {
              test: /style\.(exec\.js|js)$/i,
              use: [
                {
                  loader: require.resolve('./helpers/testLoader'),
                  options: {},
                },
                {
                  loader: path.resolve(__dirname, '../src'),
                  options: {
                    exec: true,
                  },
                },
              ],
            },
          ],
        },
      }
    );
    const stats = await compile(compiler);

    const codeFromBundle = getCodeFromBundle('style.exec.js', stats);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with "JSS" parser', async () => {
    const compiler = getCompiler(
      './jss/index.js',
      {},
      {
        module: {
          rules: [
            {
              test: /style\.js$/i,
              use: [
                {
                  loader: require.resolve('./helpers/testLoader'),
                  options: {},
                },
                {
                  loader: path.resolve(__dirname, '../src'),
                  options: {
                    postcssOptions: {
                      parser: 'postcss-js',
                    },
                  },
                },
              ],
            },
          ],
        },
      }
    );

    const stats = await compile(compiler);

    const codeFromBundle = getCodeFromBundle('style.js', stats);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
