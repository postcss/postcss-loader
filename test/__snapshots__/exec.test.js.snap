// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"exec" option should work with "Boolean" value: css 1`] = `
"a {
    color: green
}"
`;

exports[`"exec" option should work with "Boolean" value: errors 1`] = `Array []`;

exports[`"exec" option should work with "Boolean" value: warnings 1`] = `Array []`;

exports[`"exec" option should work with "JSS" parser: css 1`] = `
"a {
    color: yellow
}"
`;

exports[`"exec" option should work with "JSS" parser: errors 1`] = `Array []`;

exports[`"exec" option should work with "JSS" parser: warnings 1`] = `Array []`;
